src/sonic-yang-models/yang-models/sonic-fine-grained-ecmp.yang

```diff
@@ -0,0 +1,136 @@
+                key "name";
+
+                leaf name{
+                    type string;
+                    description "FG_NHG group name";
+                }
+
+                leaf bucket_size{
+                    type uint16;
+                    mandatory true;
+                    description "total hash bucket size desired, 
+                    recommended value of Lowest Common Multiple 
+                    of 1..{max # of next-hops}";
+                }
+
+                leaf match_mode{
+                    type string{
+                        pattern "(route-based|nexthop-based)"{
+                            error-message "Invalid FG_NHG match mode";
+                        }
+                    }
+                    mandatory true;
+                    description " The filtering method used to identify
+                    when to use Fine Grained vs regular route handling.
+                    nexthop-based looks to next-hop IP to filter routes
+                    and uses fine grained ecmp when nexthop IPs matches
+                    FG_NHG_MEMBER IPs. route-based looks to prefix to
+                    filter routes, and uses fine grained ecmp when the
+                    route prefix matches the FG_NHG_PREFIX prefix.";
+                }
+            }
+             /* end of list FG_NHG_LIST */
+        }
+        /* end of container FG_NHG */
+
+        container FG_NHG_PREFIX{
+
+            description "FG_NHG_PREFIX part of config_db.json";
+
+            list FG_NHG_PREFIX_LIST{
+
+                key "ip_prefix";
+
+                leaf ip_prefix{
+                    type stypes:sonic-ip-prefix;
+                    description "FG_NHG_PREFIX for which FG behavior is
+                    desired";
+                }
+
+                leaf FG_NHG{
+                    type string;
+                    mandatory true;
+                    description "Fine Grained next-hop group name";
+                }
+            }
+            /* end of list FG_NHG_PREFIX_LIST */
+        }
+        /* end of container FG_NHG_PREFIX */
+
+        container FG_NHG_MEMBER{
+
+            description "FG_NHG_MEMBER part of config_db.json";
+
+            list FG_NHG_MEMBER_LIST{
+
+                key "next_hop_ip";
+
+                leaf next_hop_ip{
+                    type inet:ip-address;
+                    description "FG_NHG next-hop-ip member associated
+                    with prefix";
+                }
+
+                leaf FG_NHG{
+                    type string;
+                    mandatory true;
+                    description "Fine Grained next-hop group name";
+                }
+
+                leaf bank{
+                    type uint16;
+                    mandatory true;
+                    description "An index which specifies a bank/group
+                    in which the redistribution is performed";
+                }
+
+                leaf link{
+                    type string;
```

Comment:
Needs to be a leafref to a union of sonic-port and sonic-portchannel, otherwise the type checks will only check that it is a string and not perform a check that it is a valid interface string.

Reply:
changed to union of port and portchannel